### PR TITLE
Add support for encoding abilities (fixes #28)

### LINKED ISSUE #28: Consider adding a new `MediaConfigurationType` for Real-Time v/a recording
Consider adding a new [`MediaConfigurationType`](https://wicg.github.io/media-capabilities/#enumdef-mediaconfigurationtype) to describe real-time recording.

---
How it could be done: This spec is based on the [`query(MediaConfiguration configuration)`](https://wicg.github.io/media-capabilities/#navigators-extensions); [`MediaConfiguration`](https://wicg.github.io/media-capabilities/#mediaconfiguration) has a [`type`](https://wicg.github.io/media-capabilities/#mediaconfigurationtype) with available entries: "file" and "media-source".  We could add a e.g. `media-recording` entry, which would make things look a bit like e.g.:

```
navigator.mediaCapabilities.query({
  type : "media-recording",
  video : {contentType : "video/webm", width : 1280, height : 720}
});
```
 ... returning (a Promise resolved with) an interface [`MediaDecodingAbility`](https://wicg.github.io/media-capabilities/#media-decoding-ability), 
  e.g. `{ supported : true, smooth : false, powerEfficient : true };`

---
This would provide a solution to a controversial debate in medicapture-record (see https://github.com/w3c/mediacapture-record/issues/59 and https://github.com/w3c/mediacapture-record/pull/65); @uysalere, @pehrsons since we discussed it before.

### PR DESCRIPTION
**Pull request by mounirlamouri**
@chcunningham @miguelao @uysalere PTAL :)

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -99,7 +108,7 @@ spec: mediaqueries-4; urlPrefix: https://drafts.csswg.org/mediaqueries-4/#
 </section>
 
 <section>
-  <h2 id='decoding-capabilities'>Decoding Capabilities</h2>
+  <h2 id='decoding-enconding-capabilities'>Decoding and Encoding Capabilities</h2>
```
Comment (miguelao) [inline index.bs]: nit: s/enconding/encoding/

File: index.bs [hunk 2]
```diff
@@ -126,30 +147,44 @@ spec: mediaqueries-4; urlPrefix: https://drafts.csswg.org/mediaqueries-4/#
...
         </ul>
       </p>
     </section>
 
     <section>
+      <h4 id='mediaencodingtype'>MediaEncodingType</h4>
+
+      <pre class='idl'>
+        enum MediaEncodingType {
+          "todo",
```
Comment (miguelao) [inline index.bs]: You could add already here a `"local"` entry, but I'm fine to leave it to a later PR too.

File: index.bs [hunk 3]
```diff
@@ -248,48 +283,126 @@ spec: mediaqueries-4; urlPrefix: https://drafts.csswg.org/mediaqueries-4/#
...
-      The user agent SHOULD not take into consideration the current power source
-      in order to determine the decoding power efficiency unless the device's
-      power source has side effects such as enabling different decoding modules.
+      The <dfn for='MediaCapabilitiesInfo' attribute>powerEfficient</dfn>
+      attribute MUST return <a for=MediaCapabilitiesInfo>powerEfficient</a>.
     </p>
 
     <p class='note'>
-      Authors can use {{MediaDecodingAbility/powerEfficient}} in concordance
+      Authors can use {{MediaCapabilitiesInfo/powerEfficient}} in concordance
```
Comment (miguelao) [inline index.bs]: Maybe here s/MediaCapabilitiesInfo/MediaDecodingConfiguration/ since this paragraph speaks about decoding?

File: index.bs (hunk 1 above)
Comment (mounirlamouri) [inline index.bs]: Done

File: index.bs (hunk 2 above)
Comment (mounirlamouri) [inline index.bs]: I would prefer you and @uysalere to write this :) You two should own this part.

File: index.bs (hunk 3 above)
Comment (mounirlamouri) [inline index.bs]: This is referring to the interface so we can't really use something else.

**Comment by mounirlamouri** [comment]
I'm going to merge this based on @miguelao lgtm but @chcunningham and @uysalere, feel free to file bugs if you have comments that you did not have time to leave.

**Pull request merged**